webapp/src/ts/services/version.service.ts

```diff
@@ -35,6 +38,19 @@ export class VersionService {
       });
   }
 
+  getServiceWorker ():Promise<{ version }> {
+    return new Promise((resolve, reject) => {
+      this.http
+        .get('/deploy-info.json')
```

Comment:
Another alternative that seems cleaner:
```suggestion
        .get('/deploy-info.json')
        .toPromise()
        .then(....
```

Reply:
`toPromise` is deprecated. Switched to using `firstValueFrom` instead.